tests/cases/fourslash/fourslash.ts

```diff
@@ -539,6 +548,21 @@ declare namespace FourSlashInterface {
         triggerCharacter?: string;
         isNewIdentifierLocation?: boolean;
     }
+    type ExpectedCompletionEntry = string | {
+        readonly name: string,
+        readonly source?: string,
+        readonly insertText?: string,
+        readonly replacementSpan?: Range,
+        readonly hasAction?: boolean,
+        readonly kind?: string,
+
+        // details
+        readonly text?: string,
+        readonly documentation?: string,
+        readonly sourceDisplay?: string,
+    };
+
+    type Many<T> = T | ReadonlyArray<T>;

```

Comment:
Personally, I find it confusing that "Many" might be mean "one".  I might have called it "OneOrMore".

Reply:
It could be 0 though... this is just a slightly less verbose way of writing an array. How about `ArrayOrSingle`?

Reply:
In that case, I guess I'm not sure I understand why we're not just using an array.

Reply:
In many places (such as `at`) there's usually only one element, so it saves a lot of typing. This type  is usually converted using `toArray`.

Reply:
Oh, I didn't think about the call site.  In Roslyn, we had something similar, but it was a way to avoid allocating the array, which didn't seem to be (and isn't) a concern here.  Yeah, I like `ArrayOrSingle`.